src/ui/buttons/props.js

```diff
@@ -696,6 +719,56 @@ export function normalizeButtonStyle(
   };
 }
 
+export function validateButtonMessage(
+  props: ?ButtonPropsInputs,
+  message: ButtonMessageInputs
+): ButtonMessage {
+  if (!message) {
+    throw new Error(`Expected props.message to be set`);
+  }
+
+  props = props || getDefaultButtonPropsInput();
+
+  const { amount, offer, color, position, align } = message;
+
+  if (amount !== undefined) {
+    if (typeof amount !== "number") {
+      throw new TypeError(
+        `Expected style.height to be a number, got: ${amount}`
+      );
+    }
+  }
+
+  if (offer) {
+    const invalidOffers = offer.filter(
+      (o) => values(MESSAGE_OFFER).indexOf(o) === -1
+    );
+    if (invalidOffers.length > 0) {
+      throw new Error(`Invalid offer(s): ${invalidOffers.join(",")}`);
+    }
```

Comment:
Since we only need one invalid value to be in a failed state, instead of iterating over each value with filter could this be simplified with the following?
``` js
for (const o of offer) {
    if (values(MESSAGE_OFFER).indexOf(o) === -1) {
        throw new Error(`Invalid offer(s): ${invalidOffers.join(",")}`);
    }
}
```

Reply:
@Seavenly I imagined this multiple invalid offer logging feature could be useful in some cases to save merchants a rerender to discover they have multiple invalid values instead of one.  For example, if there was a case where the merchant had a misspelling in the two offers they entered, they could learn they have to look at both values from a single error instead of having to rerender to learn about the second misspelling.  Satisfying use case?

Reply:
That seems fair. Thanks for explaining the use case.